packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1890,10 +1898,24 @@ export class ContainerRuntime
 			"Loaded",
 			undefined /* timestampMs */,
 			dataStoreContext.packagePath,
-			request?.headers,
+			headerData,
 		);
 
-		return dataStoreChannel;
+		// If the data store is tombstoned and tombstone enforcement is configured, throw an error.
+		if (
+			dataStoreContext.tombstoned &&
+			this.gcThrowOnTombstoneLoad &&
+			!headerData.allowTombstone
+		) {
+			// The requested data store is removed by gc. Create a 404 gc response exception.
+			throw responseToException(
```

Comment:
I do think we should move this into `GarbageCollector.nodeUpdated`.  Maybe we can have the telemetry `nodeUsed` function return a boolean or other object indicating whether or what to throw too.

I just don't want any chance of the logging and throwing to diverge.

Reply:
It's also inline with the InactiveObject throwing which is appealing (it's annoying to me that that code is a special case)

Reply:
I thought about that too. Only reason I kept it as-is is because I wanted the code that fails data store / blob retrieval to be in that object itself so whoever is looking at the code knows what's going on.
Good point about inactive object throwing. I missed that. Maybe we should move it into one place for now.

Reply:
If we want to keep tweaking it we could have `nodeUpdated` return an error object and callers are supposed to throw that.  Eh, not sure I really like that though